nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/record/sink/kafka/KafkaRecordSink_2_6.java

```diff
@@ -160,6 +165,7 @@ public class KafkaRecordSink_2_6 extends AbstractControllerService implements Ka
     private volatile long maxAckWaitMillis;
     private volatile String topic;
     private volatile Producer<byte[], byte[]> producer;
+    private final Queue<Future<RecordMetadata>> ackQ = new LinkedList<>();
```

Comment:
Introducing this queue as a member variable seems like it could cause issues when multiple components are using the same RecordSink. Moving this queue to a method-local variable should avoid the potential for mixing up messages from different callers.